### ISSUE TITLE
refactor: remove dependency on "rc-dialog" & "rc-switch"

### ISSUE THREAD

**Issue by pranshuchittora** [clarisights/KnitUI#114]
Hi there,
Great work on creating an awesome component lib cum design system. :clap: :confetti_ball: 

### Description
The switch has a dependency of `rc-switch` https://github.com/clarisights/KnitUI/blob/acb3221e04f788d51a3a0513c3259cd69daebab9/src/components/Switch/styledComponents.tsx#L1

Similarly, Modal too has a dependency of `rc-dialog`
https://github.com/clarisights/KnitUI/blob/fa9ab8c754f52f5997102381282027945ee9d4b6/src/components/Modal/ModalWrapper.tsx#L3

I think the components should be decoupled from such dependency.

### What can be done?
- Custom implementation can be done (could make a difference in the long run :rocket: )

Let's discuss :loudspeaker:

**Comment by Kushagra8888** [comment]
Hi @pranshuchittora , Thanks for checking out knit-ui and providing your feedback!

We used rc components in some cases for the following reasons:

* it simplified the set of features that we would need to build and maintain. Being limited on resources and time the more code we add, the more surface area we introduce for bugs and inefficiencies.
* benefiting from the updates and bug fixes that go into rc components from the community.
* the core functions that we require according to our design systems, could be implemented using rc components without over-customisation.

Would love to know your thoughts on this matter...

P.S: Curious as to how you came across knit-ui

**Comment by pranshuchittora** [comment]
Hi, @Kushagra8888 thanks for the blazing fast reply.

I completely agree, but I would like to add a few things
- I can understand that being a small team it's difficult to prioritise the overhead work required for this but working incrementally could be fruitful in future.

#### What's your opinion about implementing container components like (Grid, Card...etc)

` Curious as to how you came across knit-ui`
So I came across @clarisights a couple of months ago & I am a huge fan of design systems, UI & React so `KnitUI` attracted me & here I am :wink:

**Comment by Kushagra8888** [comment]
Certainly, if see significant value add for the time that needs to be invested, we would plan to refactor them.

We may implement some container components as well, but I'm unsure about it and would defer it to our UX.

@arorayash Would it be possible to add a list of components that we are planning to implement into knit-ui, maybe in the Readme?

**Comment by Kushagra8888** [comment]
@pranshuchittora Here's the list of things planned to be implemented: https://www.figma.com/file/o3UDrumhtDT0huZEsPnGPw/KnitUI-planning?node-id=0%3A1

**Comment by pranshuchittora** [comment]
Thanks, @Kushagra8888 sounds interesting. 
In the doc, it's `Drawer/Slider` ??
The `Drawer` is nav related & slider is for input :confused: 

And, there are few comps I would like to add.
- AppBar/NavBar
- Slider (Input)
- Progress Bars (Circular & Linear)
- Chips (Useful if filters applied on matrices)

**Comment by Kushagra8888** [comment]
Yes, you're right about Drawer and Slider being completely distinct components. Thanks for your suggestions! I'll forward them to the design team...

**Comment by pranshuchittora** [comment]
Is it possible to get some design mockups?

**Comment by Kushagra8888** [comment]
@pranshuchittora Unfortunately we don't have them now, and it might take some time document the requirements, but I'll share them with you when they are avilable...

**Comment by pranshuchittora** [comment]
No problem, I will start with Grid and other containers.

**Comment by pranshuchittora** [comment]
I think this hasn't been published on `NPM`.
Any idea abt this -> https://www.npmjs.com/package/knit.design (Might be an old one, better unpublish this :confused: )

**Comment by anshumanv** [comment]
Hey the actual library is here -  https://www.npmjs.com/package/knit-ui

Will unpublish the old one. Thanks for notifying. 👍
